internal/k8s/reconciler/gateway.go

```diff
@@ -221,22 +221,26 @@ func (g *K8sGateway) assignGatewayIPFromService(ctx context.Context, service *co
 	return nil
 }
 
-// assignGatewayIPFromPod retrieves the internal IP for the Pod and assigns
+// assignGatewayIPFromPod retrieves the internal IP for the Pods and assigns
 // it to the Gateway.
 func (g *K8sGateway) assignGatewayIPFromPod(ctx context.Context) error {
-	pod, err := g.client.PodWithLabels(ctx, utils.LabelsForGateway(g.gateway))
+	pods, err := g.client.PodsWithLabels(ctx, utils.LabelsForGateway(g.gateway))
 	if err != nil {
 		return err
 	}
 
-	if pod == nil {
-		g.status.Scheduled.NotReconciled = errors.New("pod not found")
+	if len(pods) == 0 {
+		g.status.Scheduled.NotReconciled = errors.New("pods not found")
 		return nil
 	}
 
-	if pod.Status.PodIP != "" {
-		g.serviceReady = true
-		g.addresses = append(g.addresses, pod.Status.PodIP)
+	for _, pod := range pods {
+		if pod.Status.PodIP != "" {
+			g.serviceReady = true
+			if !ipInList(g.addresses, pod.Status.PodIP) {
```

Comment:
Is the check for duplicate IPs in list necessary? I think `g.addresses` is built up from scratch here every time, and the k8s API shouldn't return the same pod twice

Reply:
I can remove it, I mostly just added it because I already was adding it for the HostIP section